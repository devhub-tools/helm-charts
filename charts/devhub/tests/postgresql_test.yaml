suite: Postgresql
templates:
  - postgresql.yaml
release:
  name: devhub
  namespace: devhub

tests:
  - it: should render postgresql with default values
    set:
      postgresql:
        enabled: true
    asserts:
      - matchSnapshot: {}

  - it: should render postgresql with custom values
    set:
      postgresql:
        enabled: true
        cluster:
          name: postgres
          instances: 1
          storage:
            size: 10Gi
            storageClass: "my-storage-class"
          sharedBuffers: "1GB"
          resources:
            requests:
              cpu: "20m"
              memory: "1Gi"
            limits:
              memory: "1Gi"
          affinity:
            tolerations:
              - key: "key"
                operator: "Equal"
                value: "value"
                effect: "NoSchedule"
          backup:
            target: "prefer-standby"
            retentionPolicy: "7d"
            barmanObjectStore:
              wal:
                compression: "bzip2"
              data:
                compression: "bzip2"

        scheduledBackup:
          enabled: true
          schedule: "0 0 * * *"

    asserts:
      - matchSnapshot: {}

  - it: doesn't render postgresql if enabled is false
    set:
      postgresql:
        enabled: false
    asserts:
      - hasDocuments:
          count: 0