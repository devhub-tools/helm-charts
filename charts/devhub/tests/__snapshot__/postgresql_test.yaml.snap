should render postgresql with custom values:
  1: |
    apiVersion: postgresql.cnpg.io/v1
    kind: Cluster
    metadata:
      annotations:
        prometheus.io/port: metrics
        prometheus.io/scheme: https
        prometheus.io/scrape: "true"
      labels:
        app.kubernetes.io/instance: devhub
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: devhub
        app.kubernetes.io/version: v2.0.2
        helm.sh/chart: devhub-2.1.1
      name: postgres
      namespace: devhub
    spec:
      affinity:
        tolerations:
          - effect: NoSchedule
            key: key
            operator: Equal
            value: value
      backup:
        barmanObjectStore:
          data:
            compression: bzip2
          wal:
            compression: bzip2
        retentionPolicy: 7d
        target: prefer-standby
      bootstrap:
        initdb:
          database: devhub
          owner: devhub
      instances: 1
      managed:
        roles:
          - createrole: true
            ensure: present
            login: true
            name: devhub
            replication: true
      postgresql:
        parameters:
          shared_buffers: 1GB
        pg_hba:
          - hostssl all all all scram-sha-256
      primaryUpdateMethod: switchover
      primaryUpdateStrategy: unsupervised
      resources:
        limits:
          memory: 1Gi
        requests:
          cpu: 20m
          memory: 1Gi
      storage:
        size: 10Gi
        storageClass: my-storage-class
  2: |
    apiVersion: postgresql.cnpg.io/v1
    kind: ScheduledBackup
    metadata:
      labels:
        app.kubernetes.io/instance: devhub
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: devhub
        app.kubernetes.io/version: v2.0.2
        helm.sh/chart: devhub-2.1.1
      name: scheduled-backup
      namespace: devhub
    spec:
      backupOwnerReference: self
      cluster:
        name: postgres
      schedule: 0 0 * * *
should render postgresql with default values:
  1: |
    apiVersion: postgresql.cnpg.io/v1
    kind: Cluster
    metadata:
      annotations:
        prometheus.io/port: metrics
        prometheus.io/scheme: https
        prometheus.io/scrape: "true"
      labels:
        app.kubernetes.io/instance: devhub
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: devhub
        app.kubernetes.io/version: v2.0.2
        helm.sh/chart: devhub-2.1.1
      name: postgres
      namespace: devhub
    spec:
      bootstrap:
        initdb:
          database: devhub
          owner: devhub
      instances: 2
      managed:
        roles:
          - createrole: true
            ensure: present
            login: true
            name: devhub
            replication: true
      postgresql:
        parameters:
          shared_buffers: 256MB
        pg_hba:
          - hostssl all all all scram-sha-256
      primaryUpdateMethod: switchover
      primaryUpdateStrategy: unsupervised
      resources:
        limits:
          memory: 1Gi
        requests:
          cpu: 20m
      storage:
        size: 10Gi
